### ISSUE TITLE
In-house Slack bot

### ISSUE THREAD

**Issue by robertquitt** [CSUA/csua-backend#21]
Extending #19 

Make a slack bot that syncs our LDAP entries with the members of a slack channel, give greeting messages, etc.

With a bot, we can do a lot more than just this, there are a number of features I can think of that would be useful. Here are a few I think of:

- Access control to channels (officers, pb, other possible private channels)

    - Will need to associate slack user <-> unix user

    - Should kick/invite users

- Info about server health

- Add/remove people from mailing lists and unix groups

    - Basically a fancy interface to the django admin site, except no password/web ui needed.

- Officer points management

    - Might have to plug into google sheets or rethink how we track these

- Print out current MOTD

There are a few resources online already for this. https://medium.com/freehunch/how-to-build-a-slack-bot-with-python-using-slack-events-api-django-under-20-minute-code-included-269c3a9bf64e seems pretty helpful. Once we get a framework down, then we can start adding more features.

**Comment by mark64** [comment]
Do we want to do all these from a slackbot? Seems like most would be better suited as a web page

**Comment by robertquitt** [comment]
Some of these actions would be well suited as being accessed from a web page, I agree. We could have the same action be called from either the slack bot or from the web page, as long as we authenticate the user. 

It would probably be easier to develop the actions by doing them as web pages first then adding interfaces to them from slack.
